NuRadioReco/modules/io/RNO_G/readRNOGDataMattak.py

```diff
@@ -724,29 +728,20 @@ def run(self):
         for dataset in self._datasets:
             dataset.setEntries((0, dataset.N()))
 
-            # read all event infos of the entier dataset (= run)
-            event_infos = dataset.eventInfo()
-            wfs = None
-
-            for idx, evtinfo in enumerate(event_infos):  # returns a list
+            # read all event infos of the entire dataset (= run)
+            for evtinfo, wf in dataset.iterate(start = 0, stop = dataset.N(),
+                                               calibrated = self._read_calibrated_data,
+                                               selectors = self._selectors,
+                                               max_entries_in_mem = self._max_in_mem):
                 event_idx += 1
 
-                self.logger.debug(f"Processing event number {event_idx} out of total {self._n_events_total}")
-                t0 = time.time()
-
-                if self._filter_event(evtinfo, event_idx):
-                    continue
+                self.logger.debug(f"Processing (filtered) event number {event_idx} out of (unfiltered) total {self._n_events_total}")
+                t0 =time.time()
```

Comment:
space is missing. Although I am not sure if that still makes sense. Because the reading of the waveforms which is the thing which takes time is now not registered by this.

Reply:
Yes I leave it out for now, I guess for a true timing we would need to time in the mattak iterator itself

Reply:
Having some kind of timing would be nice but we can add this later again.